tools/RepositoryManagement/DevHome.RepositoryManagement/ViewModels/RepositoryManagementMainPageViewModel.cs

```diff
@@ -85,38 +175,192 @@ public void HideRepository(RepositoryManagementItemViewModel repository)
+
+            var commit = repositoryWithCommit.Item2;
+
+            if (commit == Commit.DefaultCommit)
+            {
+                lineItem.HasCommitInformation = false;
+            }
+            else
+            {
+                lineItem.HasCommitInformation = true;
+                lineItem.LatestCommitAuthor = commit.Author;
+                lineItem.LatestCommitSHA = commit.SHA;
+                lineItem.MinutesSinceLatestCommit = Convert.ToInt32((DateTime.Now - commit.CommitDateTime).TotalMinutes);
+            }
+
+            lineItem.HasAConfigurationFile = repository.HasAConfigurationFile;
+            lineItem.MoreOptionsButtonAutomationName = _stringResource.GetLocalized("MoreOptionsAutomationName", repository.RepositoryName);
+
             items.Add(lineItem);
         }
 
         return items;
     }
+
+    private async Task<List<Repository>> AssignSourceControlId(List<Repository> repositories)
+    {
+        foreach (var repository in repositories)
+        {
+            Guid sourceControlGuid;
+            if (!repository.HasAssignedSourceControlProvider)
+            {
+                sourceControlGuid = await _enhanceRepositoryService.MakeRepositoryEnhanced(repository.RepositoryClonePath);
+                _dataAccessService.SetSourceControlId(repository, sourceControlGuid);
+            }
+        }
+
+        return repositories;
+    }
+
+    private List<RepositoryManagementItemViewModel> HideFilterAndSort(List<RepositoryManagementItemViewModel> repositories)
+    {
+        IEnumerable<RepositoryManagementItemViewModel> filteredAndSortedRepositories = repositories.Where(x => !x.IsHiddenFromPage);
+
+        if (!string.IsNullOrEmpty(FilterText))
+        {
+            filteredAndSortedRepositories = filteredAndSortedRepositories.Where(x => x.RepositoryName.Contains(FilterText, StringComparison.OrdinalIgnoreCase));
+        }
+
+        if (_sortTag == SortOrder.NameAscending)
+        {
+            filteredAndSortedRepositories = filteredAndSortedRepositories.OrderBy(x => x.RepositoryName);
+        }
+        else
+        {
+            filteredAndSortedRepositories = filteredAndSortedRepositories.OrderByDescending(x => x.RepositoryName);
+        }
+
+        return filteredAndSortedRepositories.ToList();
+    }
+
+    private string GetConfigurationFileIfExists(string repositoryPath)
+    {
+        var configurationDirectory = Path.Join(repositoryPath, DscHelpers.ConfigurationFolderName);
+        if (Directory.Exists(configurationDirectory))
+        {
+            var fileToUse = Directory.EnumerateFiles(configurationDirectory)
+            .Where(file => file.EndsWith(DscHelpers.ConfigurationFileYamlExtension, StringComparison.OrdinalIgnoreCase) ||
+                           file.EndsWith(DscHelpers.ConfigurationFileWingetExtension, StringComparison.OrdinalIgnoreCase))
+            .OrderByDescending(configurationFile => File.GetLastWriteTime(configurationFile))
+            .FirstOrDefault();
```

Comment:
this looks copied from tools/SetupFlow/DevHome.SetupFlow/Models/CloneRepoTask.cs

why not just add the function to DscHelpers?

Reply:
That did not occur to me.  Moved to DscHelpers.  Updated this file, and CloneRepoTask.